### PR TITLE
Use libblockdev-lvm for LVM-related functionality

### PR DESCRIPTION
**Pull request by vpodzime**
This requires the following changes:

- [x] the *libblockdev-lvm* plugin needs to be required and initialized
- [x] operations on LVs (the ``org.freedesktop.UDisks2.LogicalVolume`` interface) have to reimplemented using *libblockdev-lvm*
- [x] operations on VGs (the ``org.freedesktop.UDisks2.VolumeGroup`` interface) have to reimplemented using *libblockdev-lvm*
- [x] gathering information about PVs, VGs and LVs have to reimplemented using *libblockdev-lvm*

### CODE REVIEW

**Comment by vpodzime** [comment]
> VG operations now implemented using libblockdev-lvm with the exception of thin pool creation which requires a new function in libblockdev-lvm - see rhinstaller/libblockdev#167

Missing functionality implemented in https://github.com/rhinstaller/libblockdev/pull/168

File: modules/lvm2/udiskslinuxlogicalvolume.c [hunk 1]
```diff
@@ -545,29 +535,25 @@ handle_rename (UDisksLogicalVolume   *_volume,
...
+                                               lvrename_job_func,
+                                               &data,
+                                               NULL, /* user_data_free_func */
+                                               NULL, /* GCancellable */
+                                               &error))
     {
       g_dbus_method_invocation_return_error (invocation,
                                              UDISKS_ERROR,
                                              UDISKS_ERROR_FAILED,
                                              "Error renaming volume volume: %s",
```
Comment (vojtechtrefny) [inline modules/lvm2/udiskslinuxlogicalvolume.c]: This should probably be "Error renaming logical volume"

File: modules/lvm2/udiskslinuxlogicalvolume.c [hunk 2]
```diff
@@ -778,27 +741,25 @@ handle_activate (UDisksLogicalVolume *_volume,
...
+                                               &data,
+                                               NULL, /* user_data_free_func */
+                                               NULL, /* GCancellable */
+                                               &error))
+
     {
       g_dbus_method_invocation_return_error (invocation,
                                              UDISKS_ERROR,
                                              UDISKS_ERROR_FAILED,
                                              "Error deleting logical volume: %s",
```
Comment (vojtechtrefny) [inline modules/lvm2/udiskslinuxlogicalvolume.c]: deleting -> activating

File: modules/lvm2/udiskslinuxlogicalvolume.c [hunk 3]
```diff
@@ -878,36 +834,30 @@ handle_deactivate (UDisksLogicalVolume   *_volume,
...
+                                               lvdeactivate_job_func,
+                                               &data,
+                                               NULL, /* user_data_free_func */
+                                               NULL, /* GCancellable */
+                                               &error))
     {
       g_dbus_method_invocation_return_error (invocation,
                                              UDISKS_ERROR,
                                              UDISKS_ERROR_FAILED,
                                              "Error deleting logical volume: %s",
```
Comment (vojtechtrefny) [inline modules/lvm2/udiskslinuxlogicalvolume.c]: deleting -> deactivating

File: modules/lvm2/Makefile.am [hunk 4]
```diff
@@ -75,6 +75,7 @@ libudisks2_lvm2_la_SOURCES =                                                   \
 	udiskslvm2util.h                    udiskslvm2util.c                   \
 	udiskslinuxmanagerlvm2.h            udiskslinuxmanagerlvm2.c           \
 	udiskslvm2state.h                   udiskslvm2state.c                  \
+	jobhelpers.h						jobhelpers.c					   \
```
Comment (vpodzime) [inline modules/lvm2/Makefile.am]: Yeah, tabs vs. spaces.

File: modules/lvm2/Makefile.am [hunk 5]
```diff
@@ -75,6 +75,7 @@ libudisks2_lvm2_la_SOURCES =                                                   \
 	udiskslvm2util.h                    udiskslvm2util.c                   \
 	udiskslinuxmanagerlvm2.h            udiskslinuxmanagerlvm2.c           \
 	udiskslvm2state.h                   udiskslvm2state.c                  \
+	jobhelpers.h                        jobhelpers.c					   \
```
Comment (vojtechtrefny) [inline modules/lvm2/Makefile.am]: Weird spacing here. (Tabs vs. spaces?)

File: modules/lvm2/jobhelpers.h [hunk 6]
```diff
@@ -57,6 +58,11 @@ gboolean lvcreate_job_func (UDisksThreadedJob  *job,
                             gpointer            user_data,
                             GError            **error);
 
+gboolean lvcreate_thin_pool_job_func (UDisksThreadedJob  *job,
+                                 GCancellable       *cancellable,
```
Comment (vojtechtrefny) [inline modules/lvm2/jobhelpers.h]: Weird indentation here.

File: modules/lvm2/Makefile.am (hunk 5 above)
Comment (vpodzime) [inline modules/lvm2/Makefile.am]: Exactly. Fixed now.

File: modules/lvm2/jobhelpers.h (hunk 6 above)
Comment (vpodzime) [inline modules/lvm2/jobhelpers.h]: Fixed.

**Pull request merged**